service/history/consts/const.go

```diff
@@ -99,9 +107,17 @@ var (
 	// ErrWorkflowTaskStateInconsistent is error indicating workflow task state is inconsistent, for example there was no workflow task scheduled but buffered events are present.
 	ErrWorkflowTaskStateInconsistent = serviceerror.NewUnavailable("Workflow task state is inconsistent.")
 	// ErrResourceExhaustedBusyWorkflow is an error indicating workflow resource is exhausted and should not be retried by service handler and client
-	ErrResourceExhaustedBusyWorkflow = serviceerror.NewResourceExhausted(enumspb.RESOURCE_EXHAUSTED_CAUSE_BUSY_WORKFLOW, "Workflow is busy.")
+	ErrResourceExhaustedBusyWorkflow = serviceerror.NewResourceExhausted(
+		enumspb.RESOURCE_EXHAUSTED_CAUSE_BUSY_WORKFLOW,
+		enumspb.RESOURCE_SCOPE_SYSTEM,
```

Comment:
Namespace level IIRC

Reply:
This error is returned from mutableState cache when it fails to lock a workflow. Usually this can happen if another action on that same workflow is holding the lock and waiting for a persistence operation to finish. This can be delayed by persistence requests getting rate-limited and getting retried. But this persistence rate limit can either be system-level or namespace-level. Or it can be a delay in persistence itself. That is why I kept it as system-level.

I don't have a strong opinion here. Do you still think it is namespace level?

Reply:
I sent you a link in slack where we last discussed BUSY_WORKFLOW; it's my understanding that these should all be namespace-level errors as they occur when too many requests for the same workflow are received

Reply:
I have made the change.